bundle/set_default.go

```diff
@@ -0,0 +1,61 @@
+package bundle
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/databricks/cli/libs/diag"
+	"github.com/databricks/cli/libs/dyn"
+)
+
+type setDefault struct {
+	pattern dyn.Pattern
+	key     dyn.Path
+	value   any
+}
+
+func SetDefaultMutator(pattern dyn.Pattern, key string, value any) Mutator {
+	return &setDefault{
+		pattern: pattern,
+		key:     dyn.NewPath(dyn.Key(key)),
+		value:   value,
+	}
+}
+
+func (m *setDefault) Name() string {
+	return fmt.Sprintf("SetDefaultMutator(%v, %v, %v)", m.pattern, m.key, m.value)
+}
+
+func (m *setDefault) Apply(ctx context.Context, b *Bundle) diag.Diagnostics {
+	err := b.Config.Mutate(func(v dyn.Value) (dyn.Value, error) {
+		return dyn.MapByPattern(v, m.pattern, func(p dyn.Path, v dyn.Value) (dyn.Value, error) {
+			_, err := dyn.GetByPath(v, m.key)
+			switch {
+			case dyn.IsNoSuchKeyError(err):
+				return dyn.SetByPath(v, m.key, dyn.V(m.value))
```

Comment:
~In _some_ cases this can still return a `NoSuchKeyError`.~

~If the `Get` fails with this error on a lower-level key, then setting it can still fail.~

~Consider `foo.*.bar.qux`. If `foo.*.bar` doesn't exist, we get the `NoSuchKeyError`. But the call to `SetByPath` also fails because it assumes the path leading up to the final component exists.~

~This is probably OK with the current patterns. But it's a risk, and we should have a hermetic definition of the behavior to future-proof it.~

I now see the path is guaranteed to have a single component. Then it always works.